### DISCUSSION TITLE
[2023-12-04] Library & Engine Team Update

### DISCUSSION THREAD

**Discussion by GregoryTravis** [enso-org/enso#8447] [Weekly Demo Day]
- James: Separating Table tests from the Tests project. #8397
- Radosław: Reworking Excel support to allow for reading of big files, inferring XLS format when decoding a stream 
  #8403
- Radosław: Fix nesting of `Test.group` #8439
- Greg: Better Error Trapping in map #8307
- Cassandra: Finished design for lookup_and_replace #8322 
- Jaroslav: Multi line **chained operator syntax** #8415
- Jaroslav: Convert constants to Enso values #8416
- Jaroslav: **400x** faster with linear hashing of the hash map entries #8425
- Jaroslav: Replacing usage of Source by (Location => String) #8437
- Jaroslav: Properly report errors on duplicated constructor names #8438
- Hubert: Ensure compilation is run with a compilation lock #8395
- Hubert: Build distribution for amd64 and aarch64 MacOS #8407
- Hubert: Fixing tests teardown to avoid OOMs #8418 #8432
- Hubert: Mark some tests as flaky on Windows #8446
- Pavel: Add graalpy packages to the component directory #8351
- Pavel: project-manager can start without prior GraalVM installation #8410
- Dmitrii: Record interaction between GUI and LS #8452

**Comment by JaroslavTulach** [comment]
Performance of the `EnsoHashMap` shall no longer be a problem with #8425 

![obrazek](https://github.com/enso-org/enso/assets/26887752/81d39701-aa31-41ba-8246-efcc3a21c516)

`org_enso_benchmarks_generated_Enso_Hash_Map_100000_Enso_Incremental` achieves `4.06ms` per operation which makes it **slightly faster** than `org_enso_benchmarks_generated_Enso_Hash_Map_100000_Java_Incremental` benchmark.

![obrazek](https://github.com/enso-org/enso/assets/26887752/bd09ee10-e167-4b42-be3b-1b3756c619db)

`org_enso_benchmarks_generated_Enso_Hash_Map_100000_Enso_Replacement` has been sped up **400 times** and is now just 50% slower than the `org_enso_benchmarks_generated_Enso_Hash_Map_100000_Java_Incremental` - that's a good result given that the Java `HashMap` doesn't need to keep the referential transparency at all.
